### PR TITLE
Support Jira issue creation from managed runtime

### PR DESCRIPTION
**Pull request by aiwilliams**
While working on adding support for Jira API V2 (Jira Data Center), I was completely unaware that all my refactoring was going to have an impact on the managed runtime environment. The `src/index.ts` file had only the standard exports. After releasing `2.2.0` and updating the managed runtime, that code would no longer compile. That is when I learned that it was loading files from inside the `graph-jira` project with forms such as:

```ts
import { createJiraClient } from '@jupiterone/graph-jira/dist/jira';
import JiraClient from '@jupiterone/graph-jira/dist/jira/JiraClient';
```

This PR further refactors the code to make functions more and more composable and exports necessary constants in the `src/index.ts` to make it possible for the managed runtime to get what it needs without digging into the package's internals, and also helps express to future coders that they must be concerned with these interfaces being maintained for that codebase.

Additional tests have been added to prove that `addNewIssue()` works with Jira API V3 when provided the Atlassian Document Format (ADF) for `description`. This was entirely unclear to me until I added tests to the managed runtime to prove the ability to create a Jira issue from that project, and required that I study the `alert-rules` project to understand how it was providing a valid `description` value.

Additional testing is necessary to prove out how `addNewIssue()` will work for Jira API V2, and we may need to reconsider the `alert-rules` behavior of assuming the Jira V3 API (sending an ADF `description`). [V2 issue create](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issues/) only supports a simple string. 😳

### CODE REVIEW

**Comment by aiwilliams** [comment]
@adam-in-ict @mdaum @VDubber Please review when you get some time. There will still need to be another PR that releases these changes. Note there is a related PR https://github.com/JupiterOne/jupiter-integration-jira/pull/24 for the deployment of these changes.

File: src/utils/builders.ts [hunk 1]
```diff
@@ -1,67 +1,40 @@
...
 }
 
-// goal: give this function an array of strings, a string, or a string which is a
-// JSON-formatted array, and it figures out how to return a proper array of strings
-// this is needed because local execution for tests parses the 'project' attribute as
-// a plain string, but execution in the managed environment should provide an array
-// of strings, but some old installs might still supply just a single value string
-export function buildProjectConfigs(projects: any): ProjectConfig[] {
-  const projectConfigs: ProjectConfig[] = [];
+const EMTPY_STRING = /(^$)|(^\s+$)/;
```
Comment (VDubber) [inline src/utils/builders.ts:23]: Feels like a core util opportunity.
Something like
```
StringUtils.isBlank('value');
```
We could potentially following something like this: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isBlank-java.lang.CharSequence-

Comment (aiwilliams) [reply src/utils/builders.ts:23]: Got this advice in Slack: `if (str.trim()) // Truthy`. And some other thoughts 😆 https://jptrone.slack.com/archives/C0189BF1L8N/p1642687041010900

Comment (aiwilliams) [reply src/utils/builders.ts:23]: Going to stick with the current code for now, need to get this deployed so badly and have run into some troubles there with Lambda file size.

File: package.json [hunk 2]
```diff
@@ -28,21 +28,24 @@
     "prepack": "yarn build"
   },
   "dependencies": {
-    "@jupiterone/integration-sdk-core": "^7.4.0",
     "dotenv": "^10.0.0",
     "jira-client": "^7.1.0",
     "lodash": "^4.17.21",
```
Comment (mdaum) [inline package.json:33]: **Nitpick**: Whenever I see `lodash` as a production dependency, I always ask myself if it is truly needed, as it is large and moderately prone to vulnerabilities (though most of them end up being noise as more often than not the function(s) you use are not the subject of vulnerability). In this project's case outside of tests, the only function that is used is [camelCase](https://lodash.com/docs/4.17.15#camelCase). If we ever get annoyed with Dependabot spam and frequent version bumping, we could add a pretty remedial camelCase function and change `lodash` to a `devDependency`.

**Review by mdaum — APPROVED** [review]
looking good to me! One nitpick that need not be addressed.

**Pull request merged**